tests/30rooms/30history-visibility.pl

```diff
@@ -576,8 +576,9 @@ sub ignore_presence_for
 
 sub await_event_not_history_visibility_or_presence_for
 {
-   my ( $user, $room_id, $allowed_users ) = @_;
+   my ( $user, $room_id, $allowed_users, %params ) = @_;
    await_event_for( $user,
+      %params,
```

Comment:
you've moved `%params` from the start of the arg list to the end in `GET_new_events_for`: let's be consistent and put it at the end.